OTC-325: If there is an error when requesting CN, the state of policies should remain the same

malinowskikam opened the pull request:
Changes:
- Deleted a condition which was making every offline policy invisible after a CN request (allowing re-requesting a CN for offline policy)
- Fixed getRequestedPolicies for overview of CN when no upload date is selected 
- Streamlined and fixed error processing of CN request related code
- Deleted some unused code

[https://openimis.atlassian.net/browse/OTC-325](https://openimis.atlassian.net/browse/OTC-325)

dragos-dobre commented on app/src/main/java/org/openimis/imispolicies/OverViewPolicies.java:
```diff
@@ -531,101 +482,76 @@ public void LoginDialogBox() {
-        new Thread() {
-            public void run() {
-                runOnUiThread(() -> pd = ProgressDialog.show(OverViewPolicies.this, "", getResources().getString(R.string.Get_Control_Number)));
+    private int getControlNumber(final JSONObject order, final String SmsRequired) {
```
I see similar code in OverViewControlNumbers class. In case the code is the same and to satisfy the DRY principle, we should have the method declared once and reused multiple times.

malinowskikam commented:
Additional changes:
- Added ControlNumberService, which can be called to request and store control numbers
- Extracted CN requesting code from NotEnrolledPoliciesOverview, OverViewControlNumbers and OverViewPolicies
